src/networking/index.ts

```diff
@@ -74,22 +74,24 @@ export default class Client {
     return endpoint;
   }
 
-  request<TData>(
+  request<TData, TBody = unknown>(
     method: 'GET' | 'POST' | 'PATCH',
     url: string,
-    body?: unknown,
+    body?: TBody,
   ): Promise<Auth0Response<TData>> {
-    const options: RequestOptions = {
-      method: method,
-      headers: {
-        Accept: 'application/json',
-        'Content-Type': 'application/json',
-        'Auth0-Client': this._encodedTelemetry(),
-      },
+    const headers = new Headers();
+
+    headers.set('Accept', 'application/json');
+    headers.set('Content-Type', 'application/json');
+    headers.set('Auth0-Client', this._encodedTelemetry());
+
+    const options: RequestInit = {
+      method,
+      headers,
     };
 
     if (this.bearer) {
-      options.headers.Authorization = this.bearer;
+      headers.set('Authorization', this.bearer);
```

Comment:
Changed how we set the headers as TypeScript doesnt allow you to set the headers the way we did before.

Instead, used the recommend way to initialize Headers, then set the values.